### ISSUE TITLE
Fatal error: Call to undefined function of_get_option()

### ISSUE THREAD

**Issue by vagu71** [SimonPadbury/WBST#6]
Simon, hope you're good. 

The product archive throws this error: 

**Fatal error: Call to undefined function of_get_option() in /home/actualizestudio/public_html/testwordpressv/wp-content/themes/WBST-master/functions/woocommerce-setup.php on line 50**

**Comment by SimonPadbury** [comment]
Hi Vagu71

This was all working fine. Maybe the WooCommerce plugin has moved on.

There is nothing on line 50 that would cause this error, I think.

I can’t deal with this at this time - can you obtain afresh the content of the `woocommerce-setup.php` from WooCommerce? It may also have been updated. (It will require Bootstrap classes added in).

Simon

**Comment by SimonPadbury** [comment]
Sorry, I am not seeing this error.

**Comment by vagu71** [comment]
Hi Simon, 

This came up when I replaced woocommerce_content() in woocommerce.php with: 
if ( is_singular( 'product' ) ) {
      woocommerce_content();
  }else{
   //For ANY product archive.
   //Product taxonomy, product search or /shop landing
    woocommerce_get_template( 'archive-product.php' );
  }

This was needed to be able to edit the archive-product.php file for overriding the ux of the listing pages (it wasn't taking any changes before making this replacement in wocommerce.php).

To get around the issue, I commented out the following lines in woocommerce-setup.php: 
//echo '<div id="content" class="site-content container">';
//echo '<div id="primary" class="content-area col-sm-12 col-md-8 '.get_option('site_layout').' ">';
//echo '<main id="main" class="site-main" role="main">';

And the error went away. Not sure if this would cause a problem elsewhere though.

**Comment by vagu71** [comment]
Hi Simon, 

Whenever you get a change, please look into whether commenting out this code might cause any long term problems with the theme. 

Look forward to your reply.

Best,
V

**Comment by SimonPadbury** [comment]
Hi vagu,

This from your previous email does not make sense to me – am I missing something?

```
To get around the issue, I commented out the following lines in woocommerce-setup.php: 
//echo '

';
//echo '
';
//echo '';
```

Can you add this in my code yourself?
1. Fork my WBST
2. Edit it, to correct the error you've found (which I can't replicate, it's not showing for me)
3. Send me a Pull Request

**Comment by reececropley** [comment]
Hi Simon,

I think I understand what @vagu71. I too had the same error when I tried to extend upon the woocommerce functionality of this theme.

I replace `woocommerce_content()`

with 

`if ( is_singular( 'product' ) ) {
woocommerce_content();
}else{
//For ANY product archive.
//Product taxonomy, product search or /shop landing
woocommerce_get_template( 'archive-product.php' );
}`

and received the same error.

I then went to woocommerce_setup.php and commented out lines 49 and 50, the error then went away.

**Comment by SimonPadbury** [comment]
Hi Reece,

If that's all it takes to fix it, then OK just comment out those lines. It's been a couple of years since I build a WooCommerce site in full. I have tried to see the error message that Vagu was seeing, but it's not happening for me on my Mac AMPPS installation with debugging enabled. Vagu commented something out on Feb 20, but it wasn't clear to me what that was, so I didn't touch it. As far as I know, he didn't reply when I asked him to fork my theme and do implement his fixes in there, so that I could respond to a pull request.

My `woocommerce-setup.php` is based on a tutorial I followed some years ago, I think. With my modifications for Bootstrap compatibility. Most of my modifications were in styling the buttons etc. I did some googling just now and I couldn't find my source, as far as I can remember. These were no problems at the time; I wonder whether WooCommerce has moved on.

There are some simple WooCommerce tutorials out there, for making themes compatible. Maybe WBST should be stripped back and started again, following one of these tutorials. WBST is meant to be very simplistic.

If this is causing so much trouble for people, maybe I should just delete it. I don't have time in my life to keep going with WooCommerce compatibility, and I don't need it for any projects I'm working on these days. Sorry about that.
